hbase-server/src/main/java/org/apache/hadoop/hbase/replication/BaseReplicationEndpoint.java

```diff
@@ -69,13 +68,14 @@ public void peerConfigUpdated(ReplicationPeerConfig rpc){
   @Override
   public WALEntryFilter getWALEntryfilter() {
     ArrayList<WALEntryFilter> filters = Lists.newArrayList();
-    WALEntryFilter scopeFilter = getScopeWALEntryFilter();
-    if (scopeFilter != null) {
-      filters.add(scopeFilter);
-    }
     WALEntryFilter tableCfFilter = getNamespaceTableCfWALEntryFilter();
     if (tableCfFilter != null) {
       filters.add(tableCfFilter);
+    } else {
```

Comment:
This means when any of the wildcard kind of setting (like one NS for include) is present, a per table:cf setting wont get honoured?
To be specific say we have 2 NS, ns1 and ns2.    Now ns1 is in the include NS list.  So all tables in ns1 to be replicated. Thats it. ns2 is neither in include list nor in exclude list.  But ns2 having one table with CFs of replication scope = 1. But as per this change we will not consider those tables for replication as scopeFilter wont come in then?

Reply:
That would be the case. The given table/CF should be in the namespace passed, or in the TableCFsMap defined in the addPeer command. It does break compatibility, but it's still possible to achieve the same level of granularity by using a combination of ExcludeTableCFsMap/ExcludeNamespaces/namespaces/TableCFsMap, which is much more convenient than having to alter table schema (which also requires table to be offline).

Reply:
Any way we can avoid the compatibility break? May be a bigger change but can we think ways?  The WALEntryFilter s to act in an OR fashion?  The chain is more or less an AND model.

Reply:
Maybe come up with an extra, namespace only filter, then?